dataflow/src/org/checkerframework/dataflow/cfg/CFGVisualizer.java

```diff
@@ -47,10 +53,37 @@
      *            indicate that this information should not be output.
      * @return Possible analysis results, e.g. generated files.
      */
-    <A extends AbstractValue<A>, S extends Store<S>, T extends TransferFunction<A, S>>
     /*@Nullable*/ Map<String, Object> visualize(ControlFlowGraph cfg, Block entry,
             /*@Nullable*/ Analysis<A, S, T> analysis);
 
+    /**
+     * This is a double-direction delegate pattern interface.
+     * this method would delegate the visualize responsibility
+     * to the passed store instance.
+     * Then the store would do something and delegate the 
+     * visualize responsibility back to the CFGVisualizer.
+     * @param store
+     */
+    void visualizeStore(S store);
+
+    void visualizeStoreHeader(String classCanonicalName);
+
+    void visualizeStoreFooter();
```

Comment:
Maybe move visualizeStoreFooter to be the last method that is related to a Store, to indicate the order in which the methods are called.